update_image.pl

```diff
@@ -241,7 +247,7 @@ sub processConvergedSections {
     $sections{SBE}{out}         = "$scratch_dir/$sbe_binary_filename";
     $sections{PAYLOAD}{in}      = "$payload.bin";
     $sections{PAYLOAD}{out}     = "$scratch_dir/$payload_filename";
-    $sections{HCODE}{in}        = "$hb_binary_dir/${stop_basename}.bin";
+    $sections{HCODE}{in}        = "$hcode_dir/${stop_basename}.bin";
```

Comment:
If this was made so that if hcode_dir was set it went hcode_dir, else went to hb_binary_dir, then we could probably merge this now rather than waiting for it all to go in (it would also then be backwards/forwards compatible).

Are you able to fix it up to work with either way so we can merge this sooner rather than later?